server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -138,12 +138,9 @@ public void init() throws ServletException
   {
     super.init();
 
-    // separate client with more aggressive connection timeouts
-    // to prevent cancellations requests from blocking queries
+    // Note that httpClinetProvider is setup to return same HttpClient instance on each get() so
+    // it is same http client as that is used by parent ProxyServlet.
     broadcastClient = httpClientProvider.get();
-    broadcastClient.setConnectTimeout(CANCELLATION_TIMEOUT_MILLIS);
```

Comment:
ok, i decided to set connectTimeout to 500 ms hardcoded, it should be enough for connect and given that no one complained about it ever when it was inadvertently set to 500 ms.